src/state.ts

```diff
@@ -0,0 +1,129 @@
+import ndarray from "ndarray";
+
+/**
+ * Create a new {@link DustState} object, allocating the underlying
+ * storage
+ *
+ * @param nState Number of state variables
+ *
+ * @param nParticles Number of particles
+ */
+export function dustState(nState: number, nParticles: number): DustState {
+    const data = ndarray(new Float64Array(nState * nParticles),
+                         [nParticles, nState]);
+    return new DustState(data, nState, nParticles);
+}
+
+/**
+ * A one dimensional (slice) out of a {@link DustState} or {@link
+ * DustStateTime} object
+ */
+export interface VectorView {
+    /** An array of length 1 containing the vector length */
+    shape: number[];
```

Comment:
Why not just a number? Oh, these are the things ndarray creates with `pick`? But ndarray doesn't provide an appropriate interface itself?

Reply:
Perhaps I should add an explanatory note here. This is a subset of the interface that ndarray does provide, but its interface works for any number of dimensions. The aim here is to provide a type that will only allow a single argument when indexing (the get/set functions), but we need to also redeclare the shape part because it is used somewhere else in the support code